### PR TITLE
refactor: `version` module exports a single const struct

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit rewrites the internal `version` module that exported
various information about the current executable. Instead of exporting
several consts, we are now exporting a single const structure that 
contains all the necessary information.

This is the first step towards cleaning up how we use this information
and should allow us to use SUI to be able to patch this information
in already produced binary making it easier to cut new releases.

### CODE REVIEW

**Review by littledivy — APPROVED** [review]
LGTM. I'll open a PR with sui version patching

File: cli/standalone/binary.rs
Comment (dsherret) [inline cli/standalone/binary.rs:1]: Maybe we should add a test with deno compile using `Deno.version` in JS?

**Review by dsherret — APPROVED** [review]
LGTM

File: cli/standalone/binary.rs
Comment (bartlomieju) [reply cli/standalone/binary.rs:1]: What would it test?

Comment (dsherret) [reply cli/standalone/binary.rs:1]: Just that it doesn't explode/throw? Might not be worth it.

Comment (bartlomieju) [reply cli/standalone/binary.rs:1]: Sure, I can do that

**Pull request merged**